lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -207,6 +209,64 @@ spec = do
             [ expectFieldEqual delegation (Delegating (p2 ^. #id))
             ]
 
+    describe "STAKE_POOLS_JOIN_01x - Fee boundary values" $ do
+
+        it "STAKE_POOLS_JOIN_01x - \
+            \I can join if I have anything to delegate" $ \(ctx) -> do
+            (_, p:_) <- eventually $
+                unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+            let fee = 42
+            w <- fixtureWalletWith ctx [fee + 1]
+            joinStakePool ctx (p ^. #id) (w, "Secure Passphrase")>>= flip verify
+                [ expectResponseCode HTTP.status200
+                , expectFieldEqual status Pending
+                , expectFieldEqual direction Outgoing
+                ]
+
+        it "STAKE_POOLS_JOIN_01x - \
+            \I cannot join when I have enough fee but 0 balance to delegate" $ \ctx -> do
+            (_, p:_) <- eventually $
+                unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+            let fee = 42
+            w <- fixtureWalletWith ctx [fee]
+            r <- joinStakePool ctx (p ^. #id) (w, "Secure Passphrase")
+            print r
+            expectResponseCode HTTP.status403 r
+            expectErrorMessage errMsg403Fee r
+            -- the error msg here is a bit misleading:
+            -- I'm unable to adjust the given transaction to cover the associated fee!
+            -- In order to do so, I'd have to select one or more additional inputs,
+            -- but I can't do that without increasing the size of the transaction
+            -- beyond the acceptable limit. Note that I am only missing 42 Lovelace.
+
+            -- IMO should be changed to smth like:
+            -- "Cannot join stake-pool!
+            -- I have enough ada to cover fee in order to delegate
+            -- but 0 ada to delegate after fee is deducted"
+
+        it "STAKE_POOLS_JOIN_01x - \
+            \I cannot join if I have not enough fee to cover" $ \ctx -> do
+            (_, p:_) <- eventually $
+                unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+            let fee = 42
```

Comment:
**[suggestion]**

Move this constant to be a named test data.

**Why**

Avoid magic numbers: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#prefer-named-constants-over-magic-numbers